File: 101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        if root is None: return True
        return self.issame(root.left, root.right)
        
    def issame(self, root1: TreeNode, root2: TreeNode) -> bool:
        if root1 is None and root2 is None: 
            return True
        elif root1 is None or root2 is None: 
            return False
        elif root1.val != root2.val: 
            return False
        else:
            return self.issame(root1.left, root2.right) and self.issame(root1.right, root2.left)

File: 101.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if (root == null) return true;
        return helper(root.left, root.right);
    }

    private boolean helper(TreeNode t1, TreeNode t2) {
        if (t1 == null) return t2 == null;
        if (t2 == null) return t1 == null;
        if (t1.val != t2.val) return false;
        return helper(t1.left, t2.right) && helper(t1.right, t2.left);
    }
}